app/api/slang.py

```diff
@@ -17,47 +34,84 @@ def generate_slang():
     return ret
 
 
-def generate_slang_internal():
-    """Implement internal call to generate and return a new slang word instead of API call."""
-    # Retrieve model from the session if it's already stored there.
+def _retrieve_info_from_session(model_type: str):
```

Comment:
This function now also has the side-effect that if it's not found in the session we build something new and store it instead. That might result in some unexpected behavior if we were to implement a future function that calls _retrieve_info_from_session and it unexpectedly overwrites a non-existent different model that we erroneously expected to be in the session. We should perhaps refactor that such that we leave that fallback logic in the generate_slang_internal function (can't do it in these comments / suggestions). If it's not found in the session, warn, and then call a store_in_session function and otherwise _retrieve_info_from_session. We do of course print that it's happening, but that's only visible as long as you run it locally.

Reply:
I can make the commit for this later today if you want

Reply:
Agree

(The _retrieve ... function is indeed very specific which is why I would put an underscore if we were in a class, "do not call this unless you know what you are doing")